mod/01_core/set/all/type.rb

```diff
@@ -14,9 +15,11 @@ def type_code
 end
 
 def type_name
-  return if type_id.nil?
-  type_card = Card.fetch type_id.to_i, :skip_modules=>true, :skip_virtual=>true
-  type_card and type_card.name
+  type_card.try :name
+end
+
+def type_cardname
+  type_card.try :cardname
```

Comment:
I think this change is a good thing.  It makes all the type_* methods use a more parallel method.  I think the skip_virtual is redundant since the type has to be 'real' and have a type_id.  Before, type_card didn't do the skip_modules, but probably better if it does.  Any usage that needs the mods will auto-load them at that point.

I mainly wanted to add type_cardname, and it isn't really needed that much.  I could just do type_name.cardname.